sdk/common/src/main/java/io/opentelemetry/sdk/common/RandomHolder.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.common;
+
+import java.util.Random;
+import javax.annotation.concurrent.ThreadSafe;
+
+/** Interface that pulls the most efficient random per-platform. */
+@ThreadSafe
+public interface RandomHolder {
```

Comment:
`RandomSupplier` seems a bit more idiomatic. Actually I guess this can be a util class which returns `Supplier<Random>` rather than defining an interface

Reply:
You're speaking my (functional) language now.  Great suggestion!